src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -3928,10 +3928,11 @@ let GetInstanceMemberThisVariable (vspec: Val, expr) =
     // Skip over LAM tps. Choose 'a.
     if vspec.IsInstanceMember then
         let rec firstArg e =
-          match stripDebugPoints e with
+            match stripDebugPoints e with
             | Expr.TyLambda (_, _, b, _, _) -> firstArg b
             | Expr.TyChoose (_, b, _) -> firstArg b
             | Expr.Lambda (_, _, _, [v], _, _, _) -> Some v
+            | Expr.Const (value = Const.Zero) -> None // parser recovery one `member _`
```

Comment:
Interesting, it seems like we're looking at it after it was partially analyzed already. Could it be better to prevent checking of such members at all, like it's done in https://github.com/dotnet/fsharp/pull/15458?